system/jlib/jdebug.cpp

```diff
@@ -2246,14 +2246,19 @@ static class CMemoryUsageReporter: public Thread
         bool outofhandles = false;
 #ifdef USE_OLD_PU
         FILE* procfp = fopen("/proc/uptime", "r");
+        int matched = 0;
+        OldSystemTime = 0;
         if (procfp) {
-            fscanf(procfp, "%lf %lf\n", &dbSystemTime, &dbIdleTime);
+            matched = fscanf(procfp, "%lf %lf\n", &dbSystemTime, &dbIdleTime);
             fclose(procfp);
             outofhandles = false;
         }
         latestCPU = unsigned(100.0 - (dbIdleTime - OldIdleTime)*100.0/(dbSystemTime - OldSystemTime) + 0.5);
-        OldSystemTime = dbSystemTime;
-        OldIdleTime = dbIdleTime;
+        if (procfp && matched == 2)
```

Comment:
the constructor does differently:
```
        if (!procfp || matched == 0 || matched == EOF)
            OldSystemTime = 0;
```

I think your change is better, in that in theory scanf could return 1 and the code should probably set to 0 if != 2 as you have here.
Perhaps both version should be:
```
if (!propfp || matched != 2)
{
    OldSystemTime = 0;
    OldIdleTime = 0;
}
```